registry_module.go

```diff
@@ -45,14 +45,26 @@ type RegistryModules interface {
 	ReadVersion(ctx context.Context, moduleID RegistryModuleID, version string) (*RegistryModuleVersion, error)
 
 	// Delete a registry module
+	// Deprecated: Use DeleteByName
 	Delete(ctx context.Context, organization string, name string) error
 
 	// Delete a specific registry module provider
+	// Deprecated: Use DeleteByProvider
 	DeleteProvider(ctx context.Context, moduleID RegistryModuleID) error
 
 	// Delete a specific registry module version
+	// Deprecated: Use DeleteByVersion
 	DeleteVersion(ctx context.Context, moduleID RegistryModuleID, version string) error
 
+	// Delete a registry module by name
+	DeleteByName(ctx context.Context, module RegistryModuleDeleteByNameOrProvider) error
+
+	// Delete a specific registry module by provider
+	DeleteByProvider(ctx context.Context, module RegistryModuleDeleteByNameOrProvider, provider string) error
+
+	// Delete a specific registry module by version
+	DeleteByVersion(ctx context.Context, moduleID RegistryModuleID, version string) error
```

Comment:
I think this one might be misnamed because you're not deleting a module by version, you're just deleting a module version [by version].

Also, the method signature is identical, so it's probably OK to just change the implementation of the existing method if that is necessary.

Reply:
Fair - since versions belong to providers, I figured that was implied. But, yes, technically the specified version for the given provider of the module is deleted. So something like `DeleteProviderByVersion` would be much clearer.

Reply:
The method signatures for deleting providers by version are identical, but the original `deprecated` API call is a `POST` request and the new call is a `DELETE` request, so wouldn't that be a breaking change?

Reply:
It depends on the details of the API behavior and the deprecation. Is the new method always preferred over the deprecated method? When was the new method introduced? If we need to support the old one in TFE, would it be a good idea to attempt the new DELETE method and then fall back to the old POST method?